tonic_validate/services/litellm_service.py

```
@@ -0,0 +1,131 @@
+
+
+class LiteLLMService:
+    def __init__(
+            self,
+            encoder: Encoding,
+            model: str = "gemini/gemini-1.5-pro-latest",
+            starting_wait_time: float = 1.5,
+            max_retries: int = 12,
+            exp_delay_base: int = 2,
+    ) -> None:
+        """
+        The LiteLLMService class is a wrapper around LiteLLM client for async operations using different LLMs.
+
+        Parameters
+        ----------
+        encoder: Encoding
+            The encoding to use for token count.
+        model: str
+            The model to use for completions.
+        starting_wait_time: float
+            The starting wait time between retries.
+        max_retries: int
+            The maximum number of retries to attempt.
+        exp_delay_base: int
+            Base for exponential back off delay between retries.
+        """
+        try:
+            self.check_environment(model)
+        except Exception as e:
+            logger.error(f"Error: {str(e)}")
+            raise e
+        self.model = model
+        self.encoder = encoder
+        self.max_retries = max_retries
+        self.exp_delay_base = exp_delay_base
+        self.starting_wait_time = starting_wait_time
+        self.cache = LLMCache()
+
+    def check_environment(self, model: str) -> None:
+        if 'gemini' in model:
+            if "GEMINI_API_KEY" not in os.environ:
+                raise Exception(
+                    "GEMINI_API_KEY must be set in the environment when using Gemini"
+                )
+        if 'palm' in model:
+            if "PALM_API_KEY" not in os.environ:
+                raise Exception(
+                    "PALM_API_KEY must be set in the environment when using Palm"
+                )
+        if 'claude' in model:
+            if "ANTHROPIC_API_KEY" not in os.environ:
+                raise Exception(
+                    "ANTHROPIC_API_KEY must be set in the environment when using Claude"
+                )
```

We also need to throw an exception if none of the models match